### PR TITLE
fix circuit assertion

### PR DESCRIPTION
**Pull request by cqc-melf**
fix https://github.com/CQCL/tket/issues/746

### CODE REVIEW

**Comment by cqc-melf** [comment]
Fix the missing handling of nodes with more than one classical and more than one quantum vertices to the is_valid() function of the circuit dag.

File: tket/src/Circuit/DAGProperties.cpp [hunk 1]
```diff
@@ -231,10 +241,7 @@ bool is_valid(const DAG &G) {
       CHECK(w_out_empty);
       CHECK(c_in_ports == c_out_ports);  // bijection between in and out ports
       CHECK(q_in_ports == q_out_ports);  // bijection between in and out ports
-      // this case should only happen for measure ops
-      CHECK(
-          q_in.size() == 1 && q_out.size() == 1 && c_in.size() == 1 &&
-          c_out.size() == 1);
+      CHECK(q_in.size() == q_out.size() && c_in.size() == c_out.size());
```
Comment (yao-cqc) [inline tket/src/Circuit/DAGProperties.cpp:244]: I think this condition is implied by the:

```
      CHECK(c_in_ports == c_out_ports);  // bijection between in and out ports
      CHECK(q_in_ports == q_out_ports);  // bijection between in and out ports
```

and the previous:

```
    CHECK(q_in_ports.size() == q_in_deg);
    CHECK(q_out_ports.size() == q_out_deg);
    
    CHECK(c_in_ports.size() == c_in_deg);
    CHECK(c_out_ports.size() == c_out_deg);
```

Comment (cqc-melf) [reply tket/src/Circuit/DAGProperties.cpp]: Oh yes you are right, I forgot about the other one. I have removed that.

**Review by yao-cqc — APPROVED** [review]
Thanks!

**Pull request merged**